### DISCUSSION TITLE
x-html does not work but jQuery's html() works

### DISCUSSION THREAD

**Discussion by mikaeledstrom** [alpinejs/alpine#2838] [1. Help]
Hello,

I'm trying to refactor a page from jQuery to AlpineJS. The page currently uses jQuery both to fetch HTML snippet from an API and place the snippet inside a div. The snippet code contains a couple of external JS scripts, a couple of divs and an iframe. 

Current solution with jQuery is working
a) $.get to get the snippet code from API
b) use $('#id').html(snippet) to inject the snippet code into the correct div

Change:
a) Change $.get to fetch to get the snippet code from API. 
b) snippet code is stored in Alpine variable called snippet
c) the div which should get the snippet code has x-html="snippet".

Actual:
a) and b) works fine but nothing happens related to c). I check the inspector and no network request from the snippet code is being made. No request to load the external javascript or the iframe source. When I check the iframe contents in Inspector, the document is completely empty. The snippet code when no external JS script is running is empty. If I replace x-html with x-text, I see the whole snippet code. No errors or other messages in console either.

What can be the issue here? Any idea of possible solution?

**Comment by SimoTod** [comment]
Browsers don't run dynamically injected javascript out of the box (jquery has an hack for it).

**Reply by mikaeledstrom** [reply]
Thank you! Do you know of any other hack than jQueryfor it?

**Reply by SimoTod** [reply]
You can define your own directive: https://codepen.io/SimoTod/pen/vYpvoxv
